Funcky.Test/Extensions/ParseExtensions/ParseExtensionsTest.DateTime.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using FsCheck;
+using FsCheck.Xunit;
+using Funcky.Extensions;
+using Funcky.Monads;
+using Funcky.Xunit;
+using Xunit;
+using static Funcky.Functional;
+
+namespace Funcky.Test.Extensions
+{
+    public sealed partial class ParseExtensionsTest
+    {
+        [Theory]
+        [MemberData(nameof(DateTimeStrings))]
+        public void GivenAStringParseDateTimeOrNoneReturnsTheCorrectValue(Option<DateTime> expected, string? input)
+        {
+            using var current = new DisposableCulture("de-CH");
+
+            Assert.Equal(expected, input!.ParseDateTimeOrNone());
+        }
+
+        [Theory]
+        [MemberData(nameof(DateTimeStrings))]
+        public void TheParseOrNoneExtensionReturnTheSameValueAsTheTryParseMethods(Option<DateTime> expected, string? input)
+        {
+            using var current = new DisposableCulture("de-CH");
+
+            _ = expected;
+
+            if (DateTime.TryParse(input, out var dateTime))
+            {
+                Assert.Equal(dateTime, FunctionalAssert.IsSome(input!.ParseDateTimeOrNone()));
```

Comment:
Unnecessary null suppression with `!`. (`DateTime.TryParse` has a `[NotNullWhen(true)]` attribute on the `input` param so the compiler knows that in this branch the input is not null.)

Reply:
error CS8604: Possible null reference argument for parameter 'candidate' in 'Option<DateTime> ParseExtensions.ParseDateTimeOrNone(string candidate, IFormatProvider? provider = null, DateTimeStyles styles = DateTimeStyles.None)'.